src/test/java/DirectionsUtilityTest.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import com.google.sps.utility.DirectionsUtility;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class DirectionsUtilityTest {
+
+  private final String ORIGIN = "University of Waterloo";
+  private final String DESTINATION = "Wilfrid Laurier University";
+  private final List<String> NO_WAYPOINTS = Arrays.asList();
+  private final List<String> ONE_WAYPOINT = Arrays.asList("University of Waterloo Place");
+  private final List<String> MANY_WAYPOINTS =
+      Arrays.asList("University of Waterloo Place", "Google Kitchener");
+  private final String API_KEY = "SAMPLE_API_KEY";
```

Comment:
Some other edge cases to consider: what if one of the end points is also a waypoint, what if  a waypoint is duplicated in the many waypoints scenario, is it possible to have an unrecognized address? What are the expected behaviours for these cases? Should we test for those>

Reply:
These are all valid edge cases! All of these should be handled in the step before using this utility which involves the Google Places API (handles all the unrecognized addresses) and creates only valid origin, destination and waypoints parameters. I should definitely note this on the file, thanks!

Reply:
`RESOLVED`